backend/users/views/user.py

```diff
@@ -232,3 +243,57 @@ def generate_reset_password_token(self, request, *args, **kwargs):
         data = {"reset_password_token": jwt_token}
 
         return ApiResponse(data=data, status=status.HTTP_200_OK, headers=self.get_success_headers({}))
+
+
+class UserAwaitingApprovalViewSet(
+    GenericViewSet,
+    mixins.ListModelMixin,
+):
+    user_model = get_user_model()
+    permission_classes = [IsAdmin]
+    pagination_class = DefaultPageNumberPagination
+    serializer_class = UserAwaitingApprovalSerializer
+    ordering_fields = ["username"]
+    ordering = ["username"]
+    filter_backends = [OrderingFilter, MatchFilter, DjangoFilterBackend]
+    lookup_field = "username"
+    search_fields = ["username"]
+    filterset_class = UserFilter
+
+    def get_queryset(self):
+        return self.user_model.objects.filter(channel=None).exclude(username="deleted")
+
+    def delete(self, request, *args, **kwargs):
+        try:
+            user = User.objects.get(username=request.GET.get("username"))
+            user.delete()
+            return ApiResponse(data={"message": "User removed"}, status=status.HTTP_200_OK)
+        except User.DoesNotExist or User.MultipleObjectsReturned:
```

Comment:
I don't think we should return "User not found" if `User.MultipleObjectsReturned`. If it is currently needed to catch this exception (i.e. we don't require uniqueness on `username`), you should change the way you get users

Reply:
I'm not sure what you mean, unique username *are* required I'm catching this exception but really not expecting it (unless something *veryyyyyyy* wrong is going on with the DB)

Reply:
Then I wouldn't recommend not catching `or User.MultipleObjectsReturned`